### ISSUE TITLE
new-win-policy bug when using certain websites

### ISSUE THREAD

**Issue by dansachelarie** [nwjs/nw.js#7029]
NWJS Version : 0.37.2
Operating System : Windows

Opening some websites with 'new-win-policy' event have a strange behavior. If I do a search on www.kayak.com it will open a new page which stays stuck on "Waiting". The expected behavior is to open a new page with results

**Comment by ffanny** [comment]
I test with the code below and there is no stuck when search:
`<iframe src="https://www.kayak.com" height='400' width='700' ></iframe>`
`nw.Window.get().on('new-win-policy', function(frame, url, policy) {
  console.log("new-win-policy called: " + url);
  policy.forceNewWindow();
});`

You can provide your demo for reproducing.

**Comment by dansachelarie** [comment]
I update my test project. Type www.kayak.com and cick on Go button, do a search and you'll see 3 windows opened and one will stay on "waiting" state.
[nw-win2.zip](https://github.com/nwjs/nw.js/files/3046693/nw-win2.zip)

**Comment by ffanny** [comment]
I tried with nwjs-sdk-linux&win-v0.37.2, both work well. I guess it may be an internet problem.

<img width="951" alt="2019-04-06 (2)" src="https://user-images.githubusercontent.com/19678102/55662710-914ad180-5848-11e9-8e7a-baa3c69b2b48.png">

**Comment by dansachelarie** [comment]
No, you didn't pressed the search button to see more results. Let
![image](https://user-images.githubusercontent.com/31573257/55705128-b9703700-59e6-11e9-894a-7ac3f126fa1e.png)
 me show you my results

**Comment by ffanny** [comment]
I pressed the search button to see more results and my results are like this:
There is a window showing all the results. And a pop window if I click `view deal` button.
<img width="951" alt="2019-04-11" src="https://user-images.githubusercontent.com/19678102/55944506-54813f00-5c7b-11e9-8f94-69ffabc3c940.png">

**Comment by dansachelarie** [comment]
I really do not understand why it is not reproducing on your side. For me there are 2 tabs with result and one which stays stuck on waiting state. Just reproduced it again with v0.37.3&win. You can see bellow the second tab which is on waiting in the following picture
![image](https://user-images.githubusercontent.com/31573257/55954264-e92d7200-5c66-11e9-858c-ffffa71aff35.png)

**Comment by dansachelarie** [comment]
Let me put a video here with the problem
[kayak issue.zip](https://github.com/nwjs/nw.js/files/3068393/kayak.issue.zip)

**Comment by ffanny** [comment]
I find I will get two windows when I select the `compare vs kayak` checkbox and only one window when I select none. And waiting status will change to correct webpage after it completes loading. The load time and windows(tab) behaviors are as same as in the browser.


<img width="950" alt="2019-04-11 (3)" src="https://user-images.githubusercontent.com/19678102/55961090-aee0c680-5ca0-11e9-9954-98a9cb9c67ec.png">

**Comment by dansachelarie** [comment]
You are using the flag "--enable-features=nw2". Please remove it and retest and you'll have the same behavior. But if I use the flag the "new-win-policy" is not fired this is why I am avoid to use it. See issue https://github.com/nwjs/nw.js/issues/7012

**Comment by dansachelarie** [comment]
Any news on this one?

**Comment by ffanny** [comment]
I can reproduce this as your video shows. When I select `compare vs kayak` checkbox, there should be two windows including a kayak's result page and a compare result page. But actually, there are three windows and the `waiting` window is extra because the page of compare result we need is loaded successfully in a new window later.

**Comment by dansachelarie** [comment]
I saw that there is a new version released (nw 0.37.4) on which the new-win-policy is not triggered at all. Any idea why?

**Comment by dansachelarie** [comment]
Any news on this one? Is there anything that I can help with?

**Comment by luffs** [comment]
Found this issue when trying to use the new-win-policy event for something else.
For me, the new-win-policy event doesn't fire at all. (v0.44.1)

I went with using a webview instead of an iframe, and using this code:

```js
webview.addEventListener('newwindow', ev => {
  ev.window.discard()
  console.log('Do something with the url', ev.targetUrl)
})
```
Documentation for webview is here:
https://developer.chrome.com/apps/tags/webview#event-newwindow

Maybe new-win-policy should be deprecated and something mentioning the webview-tag added to the documentation.
